diagnostics/albedofb/albedofb.py

```diff
@@ -0,0 +1,280 @@
+#   Unless you've distirbuted your script elsewhere, you don't need to change this.
+# 
+#   Functionality
+# 
+#   Code to input many years of monthly TOA and Surface shortwave radiative fluxes
+#     and compute spatial maps of the surface albedo kernel from an isotropic model 
+#     and surface albedo 
+# 
+#   Required programming language and libraries
+# 
+#      Python3 
+# 
+#   Required model output variables
+# 
+#      TOA and Surface shortwave radiative fluxes and surface temperature
+#      ideally for 1996-2014
+# 
+#   References
+# 
+#      Donohoe, A., E. Blanchard-Wrigglesworth, A. Schweiger, and P.J. Rasch, 2020:
+#          The Effect of Atmospheric Transmissivity on Model and Observational Estimates 
+#          of the Sea Ice Albedo Feedback, J. Climate, https://doi.org/10.1175/JCLI-D-19-0674.1
+#
+
+#from __future__ import print_function
+import os
+
+# undo these for the framework version
+import matplotlib
+matplotlib.use('Agg') # non-X windows backend
+
+# Commands to load third-party libraries. Any code you don't include that's 
+# not part of your language's standard library should be listed in the 
+# settings.jsonc file.
+import xarray as xr                # python library we use to read netcdf files
+import matplotlib.pyplot as plt    # python library we use to make plots
+import xesmf as xe
+import cartopy.crs as ccrs
+import cartopy.feature as cfeature
+import albedofb_calcs
+
+
+# In[2]:
+
+
+# In[3]:
+
+
+podname='albedofb'
+
+model="{model}".format(**os.environ)
+modelname = "{model}".format(**os.environ)
+
+# these yrs only refer to the hist period for comparing kernel of mod to CERES
+firstyr = "{FIRSTYR}".format(**os.environ)
+lastyr = "{LASTYR}".format(**os.environ)
+
+wk_dir="{WK_DIR}".format(**os.environ)
+obs_dir="{OBS_DATA}/".format(**os.environ)
+output_dir = wk_dir+'/model/'
+figures_dir = wk_dir+'/model/'
+
+
+# In[4]:
+
+
+### model and obs data files and varnames: ###############################################
+
+# obs processed files, provided
+kernel_obs_file = obs_dir+'CERES40_surface_albedo_kernel_2000-2018_MJJA.nc' 
+sensitivity_obs_file = obs_dir+'CERES40_surface_sensitivity_MJJA.nc'
+
+# model output file names, to compute next
+kernel_histmod_file=output_dir+'surface_albedo_kernel_'+firstyr+'-'+lastyr+'.nc'    
+sensitivity_histmod_file=output_dir+'surface_albedo_sensitivity_'+firstyr+'-'+lastyr+'.nc' 
+
+
+# In[5]:
+
+
+# the albedo feedback modeling is done here, used same code to process CERES40 that is provided 
+albedofb_calcs.process_data(kernel_histmod_file,sensitivity_histmod_file)
+
+
+# In[6]:
+
+
+# read in kernel and albedo
+obs_dataset = xr.open_dataset(kernel_obs_file)
+obskernel = obs_dataset.kernel
+obsalbedo = obs_dataset.albedo
+
+mod_dataset = xr.open_dataset(kernel_histmod_file)
+histmodkernel = mod_dataset.kernel
+histmodalbedo = mod_dataset.albedo
+
+# set up regrid model to obs grid
+method = 'nearest_s2d'
```

Comment:
Why is `nearest_s2d` the best option here?  (Nothing wrong, just curious!)

Reply:
I've used it too long to remember why. It might be because I usually am regridding fields over the ocean only, and NaNs on land can be troublesome with some methods. This one works.

Reply:
Thanks for clarifying